katportalclient/client.py

```diff
@@ -1643,17 +1643,18 @@ def sensor_subarray_lookup(self, sub_nr, component, sensor,
 
         katcp_name: bool
             True to return the katcp name, False to return the fully qualified
-            python sensor name. Default is False.
+            Python sensor name. Default is False.
 
         Returns
         -------
         str:
             The full sensor name based on the given component and subarray.
 
         """
+        sub_nr = int(self.sitemap['sub_nr'])
```

Comment:
That's the right way to get the number out of the sitemap, but we need to check if a value was passed in before overwriting it.  Suggest:
```
if sub_nr is None:
    sub_nr = int(self.sitemap['sub_nr'])
```